src/index.js

```diff
@@ -150,12 +150,20 @@ class IOpipeWrapperClass {
   invoke() {
     this.runHook('pre:invoke');
     try {
-      return this.userFunc.call(
+      const result = this.userFunc.call(
         this.originalIdentity,
         this.event,
         this.context,
         this.callback
       );
+      if (result && result.then && result.catch) {
+        return new Promise(() => this.callback(null, () => result))
+          .then(value => value)
+          .catch(err => {
+            throw new Error(err);
```

Comment:
Why not just `throw err`?

Reply:
A new error should preserve the stack trace, and throwing a new error is a bit of a habit, having seen situations in which `throw err` causes lint complaints.  But throwing err works here, yeah.

Reply:
Huh. You're right, we definitely want to preserve the stack trace here. As long as `throw new Error()` is idomatic then I'm +1.

Reply:
`throw err` should pass the error along without losing stack trace.  Likewise, new Error preserves the trace, so...kind of a wash.  I'm inclined to leave it as `throw err`.